### PR TITLE
feat(comments): allow user to comment on specific text

### PR DESCRIPTION
**Pull request by karuhanga**
#### What does this PR do?
- add `commenting_on` field to comments
- create custom validation error handler to add message field
#### Description of Task to be completed?
- Allow users to comment on a specific text in an `article`
#### How should this be manually tested?
- Visit this [app's PR](https://ah-backend-summer-stagin-pr-47.herokuapp.com/)
- Make a request to the comments endpoint(`/api/v1/articles/:slug/comment`) with a `commenting_on` field in addition to the comment `body` field.
- This creates a comment with that specifies which part of the article you are commenting on
#### What are the relevant pivotal tracker stories?
- [#163383197](https://www.pivotaltracker.com/story/show/163383197)
#### Context
- The create comments endpoint did not previously allow a user to specify which part of the article their comment refers to
#### Screenshots
image
image

### CODE REVIEW

File: authors/apps/comments/tests/test_comments_endpoints.py [hunk 1]
```diff
@@ -240,3 +243,47 @@ def test_cannot_delete_reply_if_you_didnt_author_it(self):
             "You can only delete a reply you authored",
             response.data['response'])
         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
```
Comment (missvicki) [inline authors/apps/comments/tests/test_comments_endpoints.py]: Nice work on the tests but I have not seen tests that are passing for when you can comment on text

File: authors/apps/comments/views.py [hunk 2]
```diff
@@ -86,7 +92,7 @@ def get(self, request, pk):
         serialized_data = self.serializer_class(self.comment)
         return Response({"comment": serialized_data.data,
                          "message": (
-                    "Successfully returned details of comment: {}".format(pk)
+                             f"Successfully returned details of comment: {pk}"
```
Comment (missvicki) [inline authors/apps/comments/views.py:95]: Is the pk here same as the id? If that's the case then this return is going to be e.g "Successfully returned details of comment: 1" Could you do without the pk if that's the case, or could you return the comment body

File: authors/apps/comments/views.py [hunk 3]
```diff
@@ -95,32 +101,38 @@ def patch(self, request, pk):
...
+            self.data,
+            partial=True,
+            context={'article': self.article}
+        )
+        serializer.is_valid(raise_exception=True,
+                            message="Could not update comment")
         serializer.save()
         return Response({"comment": serializer.data,
-                    "message": "Successfully edited comment: {}".format(pk)},
+                         "message": "Successfully edited comment: {}".format(
```
Comment (missvicki) [inline authors/apps/comments/views.py:117]: Same case here, the user doesn't want to know they edited comment 1

**Review by missvicki — APPROVED** [review]

**Review by missvicki — CHANGES_REQUESTED** [review]
Previous comments

File: authors/apps/comments/tests/test_comments_endpoints.py (hunk 1 above)
Comment (karuhanga) [reply authors/apps/comments/tests/test_comments_endpoints.py]: Alright, I'll add those. Good point.

File: authors/apps/comments/views.py (hunk 2 above)
Comment (karuhanga) [reply authors/apps/comments/views.py:95]: I think we'd need a little clarification from @pbkabali on this one, my change was a basically a reformat of already existing code.

File: authors/apps/comments/views.py (hunk 3 above)
Comment (karuhanga) [reply authors/apps/comments/views.py:117]: Same as for this.

File: authors/apps/comments/views.py (hunk 2 above)
Comment (pbkabali) [reply authors/apps/comments/views.py:95]: @missvicki and guys, I made sure the message is descriptive enough for whoever is going to consume the api. In my view, there is no problem with it.

File: authors/apps/comments/tests/test_comments_endpoints.py (hunk 1 above)
Comment (karuhanga) [reply authors/apps/comments/tests/test_comments_endpoints.py]: Done.

File: authors/apps/comments/views.py (hunk 2 above)
Comment (karuhanga) [reply authors/apps/comments/views.py:95]: @missvicki does this work for you?

Comment (missvicki) [reply authors/apps/comments/views.py:95]: Still not convinced, but we can roll with it based on the discussion we had.

**Review by missvicki — APPROVED** [review]
LGTM now

**Review by malep2007 — CHANGES_REQUESTED** [review]
Could you provide a better description of this feature, especially with the steps it would take to replicate the test?

**Comment by karuhanga** [comment]
> Could you provide a better description of this feature, especially with the steps it would take to replicate the test?

Sure. I have made some changes, in particular an additional screenshot some context, an updated `README` and a slightly better description.

**Review by malep2007 — APPROVED** [review]

**Pull request merged**